ui/app/routes/auth/oidc-callback.ts

```diff
@@ -0,0 +1,33 @@
+import { CompleteOIDCAuthRequest, Ref } from 'waypoint-pb';
+
+import ApiService from 'waypoint/services/api';
+import Route from '@ember/routing/route';
+import SessionService from 'ember-simple-auth/services/session';
+import { parseResponse } from 'waypoint/authenticators/oidc';
+import { inject as service } from '@ember/service';
+
+export default class AuthIndex extends Route {
+  @service session!: SessionService;
+  @service api!: ApiService;
+
+  async model(): Promise<void> {
+    let oidcParams = parseResponse(window.location.search);
+    let completeAuthRequest = new CompleteOIDCAuthRequest();
+    completeAuthRequest.setCode(oidcParams.code);
+    let authMethodName = window.localStorage.getItem('waypointOIDCAuthMethod');
+    let authMethodRef = new Ref.AuthMethod();
+    if (authMethodName) {
+      authMethodRef.setName(authMethodName);
+    }
+    completeAuthRequest.setAuthMethod(authMethodRef);
+    completeAuthRequest.setRedirectUri(window.location.origin + window.location.pathname);
+    let nonce = window.localStorage.getItem('waypointOIDCNonce');
+    if (nonce) {
+      completeAuthRequest.setNonce(nonce);
+    }
+    completeAuthRequest.setState(oidcParams.state);
+    let resp = await this.api.client.completeOIDCAuth(completeAuthRequest, this.api.WithMeta());
+    let respObject = resp.toObject();
+    await this.session.authenticate('authenticator:oidc', respObject);
```

Comment:
Should we clean up any of the stuff in local storage at this point, or is that handled elsewhere?

Reply:
I thought about this, and then realized that because ember-simple-auth reuses the same session store and key, it is not necessary as it will overwrite the session, and a single authenticator is used https://github.com/simplabs/ember-simple-auth/blob/master/README.md?plain=1#L372-L373

Reply:
Ahhh, that makes sense 👍